Date.cpp

```diff
@@ -0,0 +1,82 @@
+#include "Date.h"
+#include <iostream>
+#include "maps.h"
+
+Date::Date(int day , int mounth) {
+    if(mounth > 12 || mounth < 1 || mounthDay.at(mounth) < day){
+        throw "Invalid Argument";
+    } else {
+        this->day = day;
+        this->mounth = mounth;       
+        this->dayOfYear = day;
+        for(int i = 1 ; i < this->mounth ; i++){
+            this->dayOfYear += mounthDay.at(i);
+        }
+        this->remainDays = DaysOfYear - this->dayOfYear;
+    }
+}
+
+Date::Date(int dayOfYear) {
+    if(dayOfYear > DaysOfYear){
+        throw "Invalid Argument";
+    } else {
+        this->dayOfYear = dayOfYear;
+        this->remainDays = DaysOfYear - dayOfYear;
+        this->calculateDate();
+    }
+}
+
+Date Date::setDate(int day , int mounth) const {
+    if(mounth > 12 || mounth < 1 || mounthDay.at(mounth) < day){
+        throw "Invalid Argument";
+    } else{
+        return Date(day , mounth);
+    }
+}
+
+Date Date::plusDay(int day) const {
+    if(day > this->remainDays || day < 0){
+        throw "Invalid Argument";
+    } else {
+        int newDay = this->dayOfYear + day;
+        return Date(newDay);
+    }
+}
+
+Date Date::minusDay(int day) const {
+    if(day > this->dayOfYear || day < 0){
+        throw "Invalid Argument";
+    } else {
+        int newDay = this->dayOfYear - day;
+        return Date(newDay);
+    }
+}

```

Comment:
These two functions still have the same logic, you can put the logic in a private function and call it in these functions.